packages/public/server/__tests-e2e__/api.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * This file is part of Serlo.org.
+ *
+ * Copyright (c) 2013-2020 Serlo Education e.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2013-2020 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/serlo.org for the canonical source repository
+ */
+import axios from 'axios'
+import * as R from 'ramda'
+
+import { testingServerUrl } from './_config'
+
+describe('/api/events', () => {
+  test('without arguments: returns list of first 100 ids', async () => {
+    const response = await fetchPath('/api/events')
+
+    expect(response.data).toEqual({
+      totalCount: 84706,
+      eventIds: R.range(1, 101),
+      pageInfo: { hasNextPage: true, hasPreviousPage: false },
+    })
+  })
+
+  test('with ?after=1000', async () => {
+    const response = await fetchPath('/api/events?after=1000')
+
+    expect(response.data).toEqual({
+      totalCount: 84706,
+      eventIds: R.range(1001, 1101),
+      pageInfo: { hasNextPage: true, hasPreviousPage: true },
+    })
+  })
+
+  test('with ?before=1000', async () => {
+    const response = await fetchPath('/api/events?before=1000')
+
+    expect(response.data).toEqual({
+      totalCount: 84706,
+      eventIds: R.range(900, 1000),
+      pageInfo: { hasNextPage: true, hasPreviousPage: true },
+    })
+  })
+
+  test('with ?first=2', async () => {
+    const response = await fetchPath('/api/events?first=2')
+
+    expect(response.data).toEqual({
+      totalCount: 84706,
+      eventIds: [1, 2],
+      pageInfo: { hasNextPage: true, hasPreviousPage: false },
+    })
+  })
```

Comment:
That is not the indented way to implement cursors. Actually, serlo.org doesn't need to be aware of the cursors in any way and just needs to return the whole list every time (api.serlo.org does the filtering etc.). Please take a look at the implementation for the notifications as an example.

Reply:
We have currently ~ 330.000 events. When we do all the filtering at api.serlo.org we need to load (or request) all events for each api call. Thus I wanted the filtering happen at serlo.org. Or do you suggest to do the filtering of `userId` and `uuid` at serlo.org and the rest at serlo.org? Then we would have a big overhead when we have no filters since we would return a list with 330.000 eventIds...

Reply:
When you think doing the filtering at api.serlo.org is no big deal, I can implement the filtering there. (I had such a solution already in https://github.com/serlo/api.serlo.org/pull/88/files/2ef840302bec1cea931110b45b5fea46ce27bb5d..493e9babfbd611f11e0c2d1760da5cf98ffa4b54 )

Reply:
IMHO:
- Filtering by userId and uuid at serlo.org level makes sense (this is also easily cacheable)
- Outputting all events might be a problem (at least if we handle the caching the same as we did for other resources at the moment). We might want to add a time-based cache to that.
- If we ran into performance problems in the future, we can still add a better cache for specific use cases (e.g. while the list of all events doesn't need up-to-date and can have a time-based cache, we might want an up-to-date list of today's events). I'd initially leave it like we have currently, though (1 request = 1 entry in cache. Smart cache value update where possible, time-based where not).